app/server.go

```diff
@@ -37,6 +37,7 @@ func main() {
 	r := mux.NewRouter()
 	r.HandleFunc("/karen/v1/users", env.PostUserHandler).Methods("POST")
 	r.HandleFunc("/karen/v1/users/auth", env.PostAuthHandler).Methods("POST")
+	r.HandleFunc("/karen/v1/users", env.GetUserHandler).Methods("GET")
```

Comment:
I know the issue says to use the `User-ID` header, but I think this API should be structured like `/karen/v1/users/{user-id}` and then use the `{user-id}` from the path to identify the user for the query. Just because one user might want to get information on another user.

Reply:
Yeah it's a weird thing because users aren't actually aware of their own `user-id` but have other people's `user-id`. Would we be able to make it `/karen/v1/users/{user-id}`, but if the `{user-id}` isn't included, just default to the `User-ID` header?

Reply:
Seems like `heimdall` [does](https://github.com/schramm-famm/heimdall/blob/fe5d97c9612606b743d79eaeff7fe6a538096cec/handlers/handlers.go#L125) return a user's ID when they call the `POST /heimdall/v1/token/auth`, so the front-end should have the current user ID stored, right?

Reply:
Oh I don't even know why that's an endpoint tbh. Since the front-end pretty much just cares about the `POST /heimdall/v1/token` handler, so it would just store the `token`. I think I put that in originally because we weren't sure how the `user-id` would be sent around.

Reply:
Oh haha, yeah I forgot that I added that as an internal API for `patches` to validate WebSocket connections, so the front-end won't be able to access it.

Reply:
Oh man that's it! I thought I was going crazy for a second there. I was like "I don't remember putting that in there, but I guess I did?" haha.

Reply:
So what about maybe updating `heimdall`'s `POST /heimdall/v1/token` route to return:
```
{
  token: "jwt_token",
  user_id: 1
}
```
so that the front-end knows the user ID. Or is that less RESTful?

Reply:
I think it just feels redundant since the token is essentially the user-id and all of our other services just use the `User-ID` header too. I feel like maybe the simpler solution would be to just make two separate routes, where `GET /karen/v1/users` would get the user's own stuff and `GET /karen/v1/users/{user-id}` would be to get someone else's, but they both use the same handler. The handler would then default to using the `User-ID` header if `{user-id}` is nil, i.e. when the `GET /karen/v1/users` route is used.

Reply:
That makes sense. What about making it `GET /karen/v1/users/me` just to be a bit more explicit?

Reply:
Lol okay how about `GET /karen/v1/users/self`?

Reply:
Sounds good!